### PR TITLE
Add open source policy

### PR DESCRIPTION
**Pull request by choldgraf**
This lays out our strategy and policy towards open source / upstream communities. It is broken up into three parts:

- **Strategy**: defines the overall strategy and principles that we follow.
- **Funding**: defines strategy/principles specifically for seeking funding.
- **Key communities**: lists a few communities we consider essential to 2i2c's mission and that we dedicate extra energy towards.

It is also paired with the blog post in the following PR, which is a brainstorm about many of these ideas and informed this policy.

- https://github.com/2i2c-org/2i2c-org.github.io/pull/138

### To do

- [x] Add something about IP (see @colliand comment below)
- [x] Leave open until ~~2022-01-20~~ 2022-01-19 to see if there are objections or iterations we need.
- [x] Fix the cross-reference to the blog post when it's posted
- [x] Post this one

### References

- closes https://github.com/2i2c-org/team-compass/issues/395

### CODE REVIEW

File: open-source/funding.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+Regardless of a specific funding opportunity, we should make our mission, strategy, budgets, and operational activities transparent so that others can assess whether our interest in funding is realistic and matched to the goals of communities we work with.
+
+## Get consent before seeking funding
+
+Before seeking funding, tell the appropriate leadership group in a community about our intentions.
+Give an idea for the scope of the funding call, what we hope to do with it, and what kind of activities it would fund (people, events, etc).
+
+## Include others in funding opportunities
+
+If possible, find at least one other community member or organization to join us in applying for funding.
```
Comment (rowanc1) [inline open-source/funding.md:38]: I think this is both the most important and hardest line! It means that everything you are doing is necessarily collaborative, and (potentially) you have less resources to manage the collaboration.

Really love the thought that has gone into this, hopefully something other non-profits and companies (e.g. curvenote) can adopt/adapt as well.

**Comment by choldgraf** [comment]
@colliand ah good point - I do think we should define some team policies/guidelines around how to handle IP. I'll add that as a task item above.

**Comment by colliand** [comment]
Policies, to the extent that they are binding, will need coordination with CS&S.

**Review by consideRatio — APPROVED** [review]
> The work you do in upstream communities does not need to be driven by 2i2c's interests or strategy. Spend your time in the way you wany to have the most impact.

Super thankful for 2i2c regarding this, and I think its critical for any organization looking to interact with an open source community to be willing to do help out like this.

I'd say its super relevant to highlight this, its such a big differentiator. I figure its the difference between self-serving focused contributions vs community-serving focused contributions.

In a section below, this is reinforced again, nice!!!

> Our upstream contributions should center the needs of upstream communities over those of our own.

And once again later! Excellent, this is being reinforced!

> We prioritize work that builds a healthy, inclusive culture in upstream communities, not just getting in technical improvements that we need for our infrastructure.

---

Super happy about this text!!!! Thank you @choldgraf and everyone iterating towards this over years of collective reflection!

File: open-source/index.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Open Source
```
[the remark this answers is not in the document]
Comment (choldgraf) [reply open-source/index.md:1]: good catch!

**Comment by choldgraf** [comment]
Thanks for the feedback everybody - I've addressed @consideRatio and @colliand's comments, I think:

- There's a new page on Intellectual Property. @colliand what do you think?
- I've also pinged CS&S on this PR to see if they have comments
- @consideRatio I made the edits you suggested in a commit
- @consideRatio do you think we need to signal-boost the message you noted more, somehow? Or is the current state enough?

cc also @yuvipanda in case he'd like to provide any thoughts, I know he's got opinions about healthy upstream interactions (and licenses :-D ).

**Comment by consideRatio** [comment]
> @consideRatio do you think we need to signal-boost the message you noted more, somehow? Or is the current state enough?

Current state is perfectly fine, it was repeated in different ways with made it prominent!

File: open-source/funding.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+When we get dedicated funding for an open source community, we act as a vehicle for the community's interests.
+We should follow the strategy and goals of the community, and consider ourselves as answerable to a community's needs.
+If it pays for a significant amount of one person's time, treat it as a temporary organization transfer[^secondment].
+
+[^secondment]: For example, [secondments](https://en.wikipedia.org/wiki/Secondment) are a process where one organization "loans" a person to another.
+
+## Carry out work in community spaces
+
+Any community-focused work should happen in community spaces, not 2i2c spaces.
```
Comment (yuvipanda) [inline open-source/funding.md:56]: YESSS This is very important!

File: open-source/ip.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+[^1]: See [the CS&S briefing on Intellectual Property](https://www.codeforsociety.org/resources/intellectual-property-in-open-projects) for an overview of some considerations here.
+
+## Software licenses
+
+### In 2i2c repositories
+
+Our fiscal sponsor holds the rights to any code that is generated in the 2i2c repository ([`@2i2c-org`](https://github.com/2i2c-org)).
+
+Any code that we control should be released under an [OSI approved license](https://opensource.org/licenses) with permissive restrictions.
+For example, the [Apache License 2.0](https://opensource.org/licenses/Apache-2.0) or the [BSD-3 Clause "New" or "Revised"](https://opensource.org/licenses/BSD-3-Clause) license. By default, we use [the BSD 3-Clause license](https://opensource.org/licenses/BSD-3-Clause).
```
Comment (yuvipanda) [inline open-source/ip.md:14]:
```suggestion
For example, the [Apache License 2.0](https://opensource.org/licenses/Apache-2.0) or the [BSD-3 Clause "New" or "Revised"](https://opensource.org/licenses/BSD-3-Clause) license. By default, we use use whatever license is most prevalent in the community where we operate in. For example, in the scientific python community, that is [the BSD 3-Clause license](https://opensource.org/licenses/BSD-3-Clause).
```

Comment (yuvipanda) [reply open-source/ip.md:14]: Be more explicit here about *why* 3-BSD, and where it is most appropriate.

**Review by yuvipanda — APPROVED** [review]
This is pretty damn good! One minor licensing suggestion otherwise <3 THANK YOU FOR WORKING ON THIS!

File: open-source/ip.md (hunk 4 above)
Comment (choldgraf) [reply open-source/ip.md:14]: good idea, thanks!

**Comment by choldgraf** [comment]
Thanks all for the reviews! I've made your suggestions as well @yuvipanda. I welcome more suggestions from others - in the meantime, here's my plan:

If I don't hear any objections, then on Jan 18th (tomorrow) I will:

- Merge this policy document
- Add a link to the blog post in https://github.com/2i2c-org/2i2c-org.github.io/pull/138
- Post the blog post and tell others about it

File: open-source/strategy.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+We should prioritize [key communities](key-communities.md) and those that [align with our values](organization:values).
+We know this will take extra effort from us - doing things on your own is faster, but doing them together with open source communities leads to a healthier and more sustianable ecosystem.
+
+## Upstream work can't be done on nights and weekends
+
+Doing work in upstream communities is central to our work, and work done with [key communities](key-communities.md) is treated as "on the clock".
+Our sustainability strategy should aim to make these upstream contributions a realistic and sustainable part of our daily activities.
+However, be mindful of the time you're spending in upstream spaces, and talk to other team members to make sure you aren't putting too much burden on them.
+
+A good balance to shoot for is:
```
Comment (damianavila) [inline open-source/strategy.md:18]: I love that we have written this down as a general policy!!

File: open-source/strategy.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+- Invite reviews of pull requests from non-2i2c team members and make sure others are on board with our proposals before taking action.
+- Put extra effort into documentation to ensure that we share context with others in the community.
+
+## Don't plant flags
+
+We are in a privileged position to have both connections to established universities and funding bodies, as well as several team members with leadership roles in open source communities.
+We should use this position to make it _easier_ for those outside of 2i2c to gain leadership roles and contribute to open source.
+
+We recognize that **2i2c is not the same thing as any open source community**, and we should avoid overpowering community dynamics just because we have access to resources.
+Our goal is to contribute and support, not to dominate.
```
Comment (damianavila) [inline open-source/strategy.md:47]: Lovely!!

**Review by damianavila — APPROVED** [review]
Lovely PR, @choldgraf!

**Comment by choldgraf** [comment]
I'll merge this now - but am happy to make edits and changes if anybody thinks we should modify something!

I also think we should consider incorporating language about **Long fixes**, after recent conversations between @yuvipanda and @consideRatio - e.g.:

```
## Make long fixes, not quick fixes

Prioritize making improvements that are long-lasting and done the "right way" over ones that quickly patch an issue but do not resolve the underlying problem. If something requires extra work to get into an upstream repository, that's fine - it'll save us time in the long-run if we fix the core issue rather than just patching things to work quickly. It will also benefit other communities that use the same tools. We should allocate team capacity to problems to give us the extra bandwidth to make these "long fixes".
```

**Pull request merged**